### PR TITLE
[wayland] allow using DRM PRIME

### PR DESCRIPTION
**Pull request by lrusak**
This doesn't do much as is unless someone compiles ffmpeg to use a v4l2 device with `AV_PIX_FMT_DRMPRIME` support. That is not tested by me though.

I'm PR'ing this now though so we can get wayland in line with gbm so in the future it's easier to add support for new features without having to add new feature support separately.

Bringing wayland inline with gbm allows for easier development (it's easier to use a window manager) and makes creating features more platform agnostic.

The `CRendererDRMPRIMEGLES` may be used by the recently added `CVideoBufferDMA` along with `CAddonVideoCodec`.

I've added the ability to read the env variable `KODI_RENDER_NODE` as currently wayland has no way to know which render device can be used with the underlying dri card. This may need to be expanded to allow easier selection for gbm but I think this is a good compromise for now.

### CODE REVIEW

**Comment by yol** [comment]
@lrusak This is GLES only, correct?

**Comment by lrusak** [comment]
> @lrusak This is GLES only, correct?

Correct (for now)

**Comment by yol** [comment]
Wouldn't it make sense to not show the setting on GL then?

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp [hunk 1]
```diff
@@ -143,13 +140,18 @@ bool CDVDVideoCodecDRMPRIME::Open(CDVDStreamInfo& hints, CDVDCodecOptions& optio
   if (pConfig && (pConfig->methods & AV_CODEC_HW_CONFIG_METHOD_HW_DEVICE_CTX) &&
       pConfig->device_type == AV_HWDEVICE_TYPE_DRM)
   {
-    CWinSystemGbm* winSystem = dynamic_cast<CWinSystemGbm*>(CServiceBroker::GetWinSystem());
-    if (av_hwdevice_ctx_create(&m_pCodecContext->hw_device_ctx, AV_HWDEVICE_TYPE_DRM,
-                               drmGetDeviceNameFromFd2(winSystem->GetDrm()->GetFileDescriptor()),
-                               nullptr, 0) < 0)
+    // todo: fix with proper device when dma-hints wayland protocol works
+    std::string device = "/dev/dri/renderD128";
```
Comment (Kwiboo) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp:144]: Please leave `drmGetDeviceNameFromFd2(winSystem->GetDrm()->GetFileDescriptor())` as the default, on my platform (with libmali) I currently only have `/dev/dri/card0` and no render device.

Comment (Kwiboo) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp:144]: ffmpeg will not use this device but it still needs it to initialize a hwdevice context.

Comment (Kwiboo) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp:144]: Or you could use a combo of `drmGetNodeTypeFromFd`, `drmGetRenderDeviceNameFromFd` and `drmGetDeviceNameFromFd2` to get the render node and fall back on `drmGetDeviceNameFromFd2` is there is no render node.

**Review by yol — CHANGES_REQUESTED** [review]
DRMPRIME setting should not be shown on GL as long as the renderer is not implemented there

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp:144]: @Kwiboo what is wrong with using the `KODI_RENDER_NODE` env variable?

**Review by yol — APPROVED** [review]

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp (hunk 1 above)
Comment (Kwiboo) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp:144]: Nothing wrong with having the `KODI_RENDER_NODE` env as an override, but your are changing the default device to  `/dev/dri/renderD128` that do not exist on all platforms where hw decoding is working today.

Comment (lrusak) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp:144]: I just would like to avoid ifdeffing it if possible

Comment (lrusak) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp:144]: @Kwiboo could you propose a solution here please?

Comment (lrusak) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp:243]: @Kwiboo ping

Comment (lrusak) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp]: @Kwiboo please look over the latest change if possible, I would like to merge soon

**Comment by lrusak** [comment]
@jernejsk I've updated this so we should avoid crashing if no render device is available.

**Comment by jernejsk** [comment]
Confirmed, it works now. Tested on LibreELEC, Allwinner H3

**Comment by lrusak** [comment]
> Confirmed, it works now. Tested on LibreELEC, Allwinner H3

Thanks!!

**Pull request merged**